src/main/java/core/basesyntax/Application.java

```diff
@@ -1,8 +1,32 @@
 package core.basesyntax;
 
+import core.basesyntax.types.CarBodyStyles;
+import core.basesyntax.types.CarBrand;
+import core.basesyntax.types.EngineManufacturers;
+import core.basesyntax.types.EngineTypes;
+
 public class Application {
+
     public static void main(String[] args) {
-        // test your code here ...
+        Engine engine = new Engine(EngineTypes.Electric, 6, 50, 2000, EngineManufacturers.Honda);
+        Car car = new Car(engine, CarBodyStyles.Buggy, CarBrand.Honda, 2010, "Blue");
+        Car carClone = car.clone();
+        Engine engineClone = engine.clone();
+
+        carClone.setYearCreate(2012);
+        carClone.setBrand(CarBrand.BMW);
+        carClone.setBodyStyle(CarBodyStyles.Minivan);
+        carClone.setColor("White");
+        carClone.setEngine(engineClone);
+
+        engineClone.setPower(2500);
+        engineClone.setEngineType(EngineTypes.Diesel);
+        engineClone.setEngineDisplacement(45.2);
+        engineClone.setEngineManufacturers(EngineManufacturers.Nissan);
+        engineClone.setCylinders(8);
```

Comment:
after you clone the car, let's take the engine of the original car and change one of its properties - this would be enough

after we print out the info for the both cars, the engines should be different